pageserver/src/tenant/timeline.rs

```diff
@@ -3511,7 +3523,7 @@ impl Timeline {
         let mut layer_names_to_delete = Vec::with_capacity(deltas_to_compact.len());
         for l in deltas_to_compact {
             layer_names_to_delete.push(l.filename());
-            self.delete_historic_layer(layer_removal_cs, l, &mut updates)?;
+            self.delete_historic_layer(layer_removal_cs.clone(), l, &mut updates)?;
```

Comment:
NIT: I think using reference instead of an owned Arc should be enough `&layer_removal_cs`. Not really important, just needless clone

Reply:
Sometimes we need to pass this into inner functions so I keep all these functions accepting `Arc<MutexGuard>`, though this one seems like a leaf call 🤣

Reply:
Yep, since this is a leaf call there is no need to clone the Arc once more